data/abilities.js

```diff
@@ -131,7 +131,7 @@ let BattleAbilities = {
 		shortDesc: "If this Pokemon (not its substitute) takes a critical hit, its Attack is raised 12 stages.",
 		onHit(target, source, move) {
 			if (!target.hp) return;
-			if (move && move.effectType === 'Move' && move.crit) {
+			if (move && move.effectType === 'Move' && this.getHitData(target).crit) {
```

Comment:
This always seemed really awkward to me.

Then it hit me: Why _isn't_ this a property of `move`? That seems like a better place for hit data than even more temporary data in `Battle`.

Reply:
For that matter, I wonder if it makes more sense to have a different `moveCopy` object for each target.

Most move state isn't shared in spread moves, right?

Reply:
Actually, "different ActiveMove for each target" sounds like a significantly nicer solution.

Reply:
Oops, I meant to reply to this.

Reply:
Also, most move-state _is_ shared between targets. Literally the only things not shared are `crit` and `typeMod`.

Reply:
Accuracy, base power, etc isn't shared either, right?

Reply:
What move state _is_ shared?

Reply:
We don't store either of those. We calculate them once per target, then throw them away because they aren't needed any more. What we do store is the following, and all of these are shared per target (this excludes things only relevant to single target moves, such as `ActiveMove#allies` which only matters for Beat Up):
- `ability`
- `aerilateBoosted`
- `auraBooster`
- `galvanizeBoosted`
- `hasAuraBreak`
- `hasBounced`
- `hasSheerForce`
- `isExternal`
- `magnitude`
- `normalizeBoosted`
- `pixilateBoosted`
- `pranksterBoosted`
- `refrigerteBoosted`
- `selfDropped`
- `spreadHit` (obviously)
- `stab`

Reply:
Oh, `zBrokeProtect` is another thing that's not shared between targets, so I'll add that to this PR.